modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskTriggerHandler.java

```diff
@@ -105,6 +107,8 @@ public final void registerHandlerFor(String subject) {
 
     @Override
     public void handle(MotechEvent event) throws TriggerNotFoundException {
+        LOGGER.debug("Handling the motech event with subject: {}", event.getSubject());
```

Comment:
Moreover we should log what is the outcome of the ifs below on info - whether we are executing or skipping, and info for each action being executed